### PR TITLE
made download links url save

### PR DESCRIPTION
**Pull request by mloidl**
Hi @wilriker,

Thank you very much for this script. 
Unfortunately it did not work for me because of white-spaces in some filenames.

So I've replaced all occurrences of `html.EscapeString` with `url.PathEscape`.

### CODE REVIEW

**Comment by wilriker** [comment]
First of all: thanks!
Second: you were right that `html.EscapeString` was indeed the wrong function. But `url.PathEscape` also only is almost the right solution. Since the path is passed in a query parameter `url.QueryEscape` would be correct. Could you please change that and then I can accept your PR.

File: duetbackup.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func getFileList(baseURL string, dir string, first uint64) (*filelist, error) {
 
 func updateLocalFiles(baseURL string, fl *filelist, outDir string) error {
 
-	fileDownloadURL := "rr_download?name=" + html.EscapeString(fl.Dir) + "/"
+	fileDownloadURL := "rr_download?name=" + url.PathEscape(fl.Dir) + "/"
```
Comment (wilriker) [inline duetbackup.go]: Please also include the slash inside `url.QueryEscape`. I was missing that one even in the original implementation.

Comment (mloidl) [reply duetbackup.go]: Thank you.
Sorry but I've stopped reading the url-doc when i had a working solution.

Comment (mloidl) [reply duetbackup.go]: slash is now escaped too.

Comment (wilriker) [reply duetbackup.go]: No worries - I basically did the same. :-D

**Pull request merged**